sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java

```diff
@@ -22,36 +22,66 @@
 
 package io.crate.protocols.postgres;
 
+import com.carrotsearch.hppc.IntHashSet;
+import com.carrotsearch.hppc.IntSet;
 import io.crate.action.sql.SQLOperations;
 import org.elasticsearch.common.component.AbstractLifecycleComponent;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.network.NetworkAddress;
+import org.elasticsearch.common.network.NetworkService;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.BoundTransportAddress;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.transport.PortsRange;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.transport.BindTransportException;
 import org.jboss.netty.bootstrap.ServerBootstrap;
+import org.jboss.netty.channel.Channel;
 import org.jboss.netty.channel.ChannelPipeline;
 import org.jboss.netty.channel.ChannelPipelineFactory;
 import org.jboss.netty.channel.Channels;
 import org.jboss.netty.channel.socket.nio.NioServerSocketChannelFactory;
 
+import java.io.IOException;
+import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
 import static org.elasticsearch.common.util.concurrent.EsExecutors.daemonThreadFactory;
 
 @Singleton
 public class PostgresNetty extends AbstractLifecycleComponent {
 
     private final SQLOperations sqlOperations;
+    private final NetworkService networkService;
+
+    private final String port;
+    private final String[] bindHosts;
+    private final String[] publishHosts;
+
     private ServerBootstrap bootstrap;
     private ExecutorService bossExecutor;
     private ExecutorService workerExecutor;
+    private BoundTransportAddress boundAddress;
+
+    private volatile List<Channel> serverChannels = new ArrayList<>();
 
     @Inject
-    public PostgresNetty(Settings settings, SQLOperations sqlOperations) {
+    public PostgresNetty(Settings settings, SQLOperations sqlOperations, NetworkService networkService) {
         super(settings);
         this.sqlOperations = sqlOperations;
+        this.networkService = networkService;
+
+        port = settings.get("psql.netty.port", settings.get("psql.port"));
```

Comment:
I think we should also register these settings as `CrateSettings` and you can use the `Int/String-Setting`  classes